rosbridge_library/src/rosbridge_library/protocol.py

```diff
@@ -46,15 +46,26 @@ def is_number(s):
         return True
     except ValueError:
         return False
-        
-def has_binary(d):
-    if type(d)==bson.Binary:
+
+
+def has_binary(obj):
+    """ Returns True if obj is a binary or contains a binary attribute
+    """
+
+    if type(obj) is bson.binary.Binary:
         return True
-    if type(d)==dict:
-        for k,v in d.items():
+
+    if type(obj) is list:
+        for item in obj:
+            if has_binary(item):
+                return True
+
+    if type(obj) is dict:
+        for k, v in obj.iteritems():
             if has_binary(v):
```

Comment:
You could simplify both this and the `list` case using `any()` and a generator expression:
```
# list case
return any((has_binary(item) for item in obj))
# dict case
return any((has_binary(item) for item in obj.itervalues()))
```